### DISCUSSION TITLE
Using NES adapter for 8bitdo M30 on NT Mini?

### DISCUSSION THREAD

**Discussion by Schmullus1** [darthcloud/BlueRetro#161] [Q&A] [answered]
Built a few adapters (Dreamcast, N64, and Gamecube so far) and have loved them, and I realized that perhaps this could be the solution to a problem that's existed for a while on the Analogue NT Mini. The jailbreak includes a Sega Genesis FPGA core, but the button mapping on an NES controller is less than ideal, with one of the 3 face buttons being mapped to Select. Pairing an 8bitdo M30 to the system seems to be equally problematic with stock firmware for their retro receivers.

What I'm wondering is, since the NES and SNES used the same kind of controller bus setup, and someone on Reddit has stated that they got SNES controllers to work (both wired and via 8bitdo adapters) with the Genesis core by using a Raphnet SNES to NES adapter, does it seem possible that I could build an NES-port equipped BlueRetro adapter, flash it to the SNES config, and then custom map a paired M30 so that the console thinks the BlueRetro is a SNES controller, and thus would be able to allow the Genesis core to finally have a decent controller setup?

From Reddit, the mapping for the core to SNES controller is as such:

Genesis - SNES
A____________B
B____________Y
C____________A
X____________X
Y____________L
Z____________R

Appreciate any insight anyone might have on this. While building the NES adapter isn't a particularly costly build, I'm not sure how to best approach the firmware config for it so I wanted to get some thoughts before jumping on it.

**Comment by darthcloud** [comment, marked as answer]
In your case yes the best is to build NES adapter and flash the SNES firmware.

Note that BlueRetro is build for original console which typically poll controller once every frame and have a quite relaxed timing on the data read after clk is set.

Adapter often deviate from this and poll continuously and assume TTL chip fast data transition. In those case BlueRetro wouldn't work at all.

**Reply by Schmullus1** [reply]
Awesome, glad to hear it. I know it's possible to directly wire a SNES controller into an NES plug and use it on the original hardware with no other conversions in the middle, so that's why I had hope that this would probably work. Just ordered the parts I need so I'll follow up if everything works out in case anyone ends up looking to do the same thing down the road.

**Reply by Schmullus1** [reply]
As promised, circling back to this now that I have it all working to confirm that it was a success. Took a bit of trial and error to figure out a working config, but I settled on the following:

- Build an NES adapter flashed with SNES firmware. I built a single controller plug version for now but plan to convert it to a 2-player adapter later. The NT Mini has a built-in Four Score so I'm not sure if this might cause any issues. Worst case, build two adapters. It cost me $25 total for the parts to build two of these, so it's not too bad in the long run IMO.
- Pair M30 to adapter using X-Input mode.
- Set controller mapping to the following options:
Src Label: Xbox One S / X|S
Dst Label: SNES

Left stick left - D-Pad left
Left stick right - D-Pad right
Left stick down - D-Pad down
Left stick up - D-Pad Up
X - X
B - Y
A - B
Y - L
Menu - Start
View - Select (Acts as the Genesis 6-Button Mode Select Button)
RT - A
RB - R

This makes the buttons all match their Genesis counterparts, with the middle button (-) of the added trio on the M30 acting as the mode switch for games that require the controller to only appear as a 3-button. I haven't tested the mode switch but it is detected by the controller test ROM I tried so it should be fine.

**Reply by darthcloud** [reply]
Glad it work! Enjoy!
